ActiveDataProvider.php

```diff
@@ -76,6 +76,31 @@ public function getAggregation($name)
         return $aggregations[$name];
     }
 
+    /**
+     * @return array all suggestions results
+     */
+    public function getSuggestions()
+    {
+        $results = $this->getQueryResults();
+        return isset($results['suggest']) ? $results['suggest'] : [];
+    }
+
+    /**
+     * Returns results of the specified suggestion.
+     * @param string $name suggestion name.
+     * @return array suggestion results.
+     * @throws InvalidCallException if query results do not contain the requested suggestion.
+     */
+    public function getSuggestion($name)
+    {
+        $suggestions = $this->getSuggestions();
+        Yii::info($suggestions);
+        if (!isset($suggestions[$name])) {
+            throw new InvalidCallException("Suggestions '{$name}' not found.");
```

Comment:
How about `Suggestion '{$name}' not found`?

Reply:
Yes this is a better error message.